tests/test_search_api_client.py

```diff
@@ -0,0 +1,380 @@
+# -*- coding: utf-8 -*-
+import os
+
+from flask import json
+import requests_mock
+import pytest
+import mock
+
+from dmapiclient import SearchAPIClient
+from dmapiclient import APIError, HTTPError
+
+
+@pytest.yield_fixture
+def rmock():
+    with requests_mock.mock() as rmock:
+        real_register_uri = rmock.register_uri
+
+        def register_uri_with_complete_qs(*args, **kwargs):
+            if 'complete_qs' not in kwargs:
+                kwargs['complete_qs'] = True
+
+            return real_register_uri(*args, **kwargs)
+
+        rmock.register_uri = register_uri_with_complete_qs
+
+        yield rmock
+
+
+@pytest.fixture
+def search_client():
+    return SearchAPIClient('http://baseurl', 'auth-token', True)
+
+
+@pytest.fixture
+def service():
+    """A stripped down G6-IaaS service"""
+    return {
+        "id": "1234567890123456",
+        "supplierId": 1,
+        "lot": "IaaS",
+        "title": "My Iaas Service",
+        "lastUpdated": "2014-12-23T14:46:17Z",
+        "lastUpdatedByEmail": "supplier@digital.cabinet-office.gov.uk",
+        "lastCompleted": "2014-12-23T14:46:22Z",
+        "lastCompletedByEmail": "supplier@digital.cabinet-office.gov.uk",
+        "serviceTypes": [
+            "Compute",
+            "Storage"
+        ],
+        "serviceName": "My Iaas Service",
+        "serviceSummary": "IaaS Service Summary",
+        "serviceBenefits": [
+            "Free lollipop to every 10th customer",
+            "It's just lovely"
+        ],
+        "serviceFeatures": [
+            "[To be completed]",
+            "This is my second \"feture\""
+        ],
+        "minimumContractPeriod": "Month",
+        "terminationCost": True,
+        "priceInterval": "",
+        "trialOption": True,
+        "priceUnit": "Person",
+        "educationPricing": True,
+        "vatIncluded": False,
+        "priceString": "£10.0067 per person",
+        "priceMin": 10.0067,
```

Comment:
Is that a valid price?

Reply:
No idea - I didn't change any of this, I just moved it 😄

Reply:
Yeah sorry, just realised it!

Reply:
Yes that's a valid price.

For G-Cloud things you might have like "0.0067p per GB" or something.  There was a user need - in that suppliers complained *a lot* when we restricted it to 2 decimal places, I think during G-7 applications, and we had to release a fix for it quickly, during the application period.

Reply:
e.g. this looks wrong, but is a valid pricing structure for G-Cloud: https://www.digitalmarketplace.service.gov.uk/g-cloud/services/911824080513785

Reply:
Ah yeah, that makes sense